File: Data Structures & Algorithms/search-2d-matrix/submission-3.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int mn = matrix.length * matrix[0].length;
        int m = matrix.length;
        int n = matrix[0].length;
        int left = 0;
        int right = mn - 1;

        while(left <= right){
            int mid = (right - left)/2 + left;

            if(matrix[mid / n][mid % n] == target){
                return true;
            }
            else if(matrix[mid / n][mid % n] < target){
                left = mid + 1;
            }
            else{
                right = mid - 1;
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-3.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int left = 0;
        int right = nums.size() - 1;

        //int min = INT_MAX;

        while(left <= right){
            int mid = (right - left)/2 + left;
            //min = std::min(min, nums[mid]);
            if(nums[mid] >= nums[right]){
                left = mid + 1;
            }
            else{
                right = mid;
            }
        }

        return nums[right];
    }
};
